src/main/scala/chiseltest/internal/BackendExecutive.scala

```diff
@@ -1,24 +1,56 @@
 // SPDX-License-Identifier: Apache-2.0
 
-package chiseltest.backends
+package chiseltest.internal
 
-import chiseltest.internal.BackendInstance
-import chisel3.experimental.BaseModule
+import chisel3.experimental.{BaseModule, DataMirror}
 import chisel3.internal.firrtl.Circuit
 import chisel3.{Data, Element, Module, Record, Vec}
+import chiseltest.coverage.Coverage
+import chiseltest.simulator.Simulator
 import firrtl.AnnotationSeq
 import firrtl.annotations.ReferenceTarget
-import firrtl.transforms.CombinationalPath
+import firrtl.transforms.{CheckCombLoops, CombinationalPath}
 
-trait BackendExecutive {
-  def getTopModule(circuit: Circuit): BaseModule = {
+object BackendExecutive {
```

Comment:
@ducky64 I changed this from a `trait` to an `object` since it is generic for all `Simulator` backends now. OK?

Reply:
Huh? I'm confused.

Don't the methods in BackendExecutive depend on circuit state? Doesn't an object mean you can only have one of these (singleton)? What gain are we hoping for here?

Reply:
> Don't the methods in BackendExecutive depend on circuit state?

Only the `BackendInstance` depends on the circuit state. The `BackendExecutive` just contains a function that lets you create a `BackendInstance` from a Chisel module. Since it only contains functions, making it an `object` makes sense to me.